tstest/integration/testcontrol/testcontrol.go

```diff
@@ -293,20 +296,24 @@ func (s *Server) getUser(nodeKey tailcfg.NodeKey) (*tailcfg.User, *tailcfg.Login
 	if u, ok := s.users[nodeKey]; ok {
 		return u, s.logins[nodeKey]
 	}
-	id := tailcfg.UserID(len(s.users) + 1)
+	userID := tailcfg.UserID(len(s.users) + 1)
+	if s.AllNodesSameUser {
+		userID = 42
+	}
```

Comment:
This is a hacky way to make all of the users in integration tests have the same id. For some reason it causes an overwrite in a map somewhere, resulting in unknown target, not in your tailnet